gcsio/src/main/java/com/google/cloud/hadoop/gcsio/EventLoggingHttpRequestInitializer.java

```diff
@@ -0,0 +1,89 @@
+package com.google.cloud.hadoop.gcsio;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.HttpResponse;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.MapMaker;
+import com.google.common.flogger.GoogleLogger;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+class EventLoggingHttpRequestInitializer implements HttpRequestInitializer {
+  private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+  private static final Gson gson = new Gson();
+
+  // Using a ConcurrentHashMap with weak key reference to avoid potential memory leak during failure
+  // scenarios. We do not get an response interceptor callback if there is some failure while
+  // executing the
+  // HttpRequest.
+  private final Map<HttpRequest, HttpRequestResponseTimeTracker> requestTracker =
```

Comment:
static?

Reply:
Static also would have worked as well. I think having non-static is more natural here since there will be on tracker per GCSCloudStorage.